index.js

```diff
@@ -35,6 +35,9 @@ module.exports = function(str, options) {
   }
 
   var lines = str.match(re) || [];
+  //strip any remaining newline characters at the end of the lines, these could
+  //be there due to newline characters in the input text
+  lines = lines.map(function (line) {return line.replace(/(\r\n|\n|\r)$/gm,'')});
```

Comment:
sorry for the really late reply, I missed this pr somehow. This probably isn't the best solution. I'm not sure what is though, I'll have to try some things out and report back